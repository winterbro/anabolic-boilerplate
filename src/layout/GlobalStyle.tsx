import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  :root {
    --sizing-unit: 8px;
    --space-xs: var(--sizing-unit);
    --space-sm: calc(2 * var(--sizing-unit));
    --space-md: calc(3 * var(--sizing-unit));
    --space-lg: calc(4 * var(--sizing-unit));
    --space-xl: calc(6 * var(--sizing-unit));
    --font-family: "Lato", sans-serif;
    --font-size: calc(2 * var(--sizing-unit));
  }

  * {
    box-sizing: border-box;
    padding: 0;
    margin: 0;
  }

  html {
    font-family: var(--font-family);
    font-size: var(--font-size);
  }

  body {
    color: #fff;
    background: #242424;
  }

  #root {
    width: 100%;
    height: 100%;
  }
`;

export default GlobalStyle;
