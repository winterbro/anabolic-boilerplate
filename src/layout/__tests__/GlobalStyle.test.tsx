import { describe, it } from 'vitest';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import GlobalStyle from 'layout/GlobalStyle';

describe('GlobalStyle', () => {
  it('should have global style', () => {
    renderer.create(<GlobalStyle />);
    expect(document.head).toMatchSnapshot();
  });
});
