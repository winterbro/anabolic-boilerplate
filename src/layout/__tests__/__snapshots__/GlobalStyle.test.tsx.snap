// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GlobalStyle > should have global style 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.3.11"
  />
</head>
`;
